syntax = "proto3";

package ozonmp.cnm_purchase_api.v1;

import "validate/validate.proto";
import "google/api/annotations.proto";
import "google/protobuf/timestamp.proto";

option go_package = "github.com/ozonmp/cnm-purchase-api/pkg/cnm-purchase-api;cnm_purchase_api";

// CnmPurchaseApiService - Service for working with purchases
service CnmPurchaseApiService {
  // DescribePurchaseV1 - Describe a purchase
  rpc DescribePurchaseV1(DescribePurchaseV1Request) returns (DescribePurchaseV1Response) {
    option (google.api.http) = {
      get: "/v1/purchases/{purchase_id}"
    };
  }
}

message Purchase {
  uint64 id  = 1;
  uint64 foo = 2;
  google.protobuf.Timestamp created = 3;
}

message DescribePurchaseV1Request {
  uint64 purchase_id = 1 [(validate.rules).uint64.gt = 0];
}

message DescribePurchaseV1Response {
  Purchase value = 1;
}
